optimum/exporters/onnx/convert.py

```diff
@@ -568,22 +571,47 @@ def remap(value):
             output_names = list(config.outputs.keys())
 
             if no_dynamic_axes:
-                dynamix_axes = None
+                dynamic_axes = None
             else:
-                dynamix_axes = dict(chain(inputs.items(), config.outputs.items()))
-
-            # Export can work with named args but the dict containing named args has to be the last element of the args
-            # tuple.
-            onnx_export(
-                model,
-                (dummy_inputs,),
-                f=output.as_posix(),
-                input_names=input_names,
-                output_names=output_names,
-                dynamic_axes=dynamix_axes,
-                do_constant_folding=do_constant_folding,
-                opset_version=opset,
-            )
+                dynamic_axes = dict(chain(inputs.items(), config.outputs.items()))
+
+            if not dynamo:
+                # Export can work with named args but the dict containing named args has to be the last element of the args
+                # tuple.
+                onnx_export(
+                    model,
+                    (dummy_inputs,),
+                    f=output.as_posix(),
+                    input_names=input_names,
+                    output_names=output_names,
+                    dynamic_axes=dynamic_axes,
+                    do_constant_folding=do_constant_folding,
+                    opset_version=opset,
+                )
+            else:
+                try:
+                    from torch.onnx import dynamo_export as dynamo_onnx_export
+                except ImportError:
+                    raise MinimumVersionError(
+                        "The dynamo export feature is only available in PyTorch >= 2.1."
+                    )
+                export_options = torch.onnx.ExportOptions(
+                    dynamic_shapes=dynamic_axes is not None,
+                )
+                if opset != 18:
+                    logger.warning(
+                        "Dynamo ONNX export only supports opset 18 for now. The opset will be set to 18."
+                    )
+                # Args and kwargs are supported natively in dynamo onnx export.
+                onnx_program = dynamo_onnx_export(
+                    model,
+                    export_options = export_options,
+                    **dummy_inputs,
+                )
+                # TODO: Much of the later code performing external data clean-up is
+                # unnecessary for dynamo onnx export. ModelProto is directly accessible
+                # from onnx_program.
+                onnx_program.save(output.as_posix())
```

Comment:
I am not familiar with the ExportedProgram -> ONNX logic (if that's the flow). How are things like `input_names`, `output_names`, `dynamic_axes`, `do_constant_folding` handled?

Reply:
Good point, in dynamo onnx export we want to simplify the interface design. In the latest version, all four things mentioned are not configurable and are handled as such:
- `input_names` and `output_names`: follows whatever names created by dynamo. 
- `dynamic_axes`: inherits the symbols created by dynamo and corresponding symbolic shape apis.
- `do_constant_folding`: not applied yet, but soon to come.
Are there particular use cases that must require custom configurations through these arguments?

Reply:
Thank you! I think `input_names` & `output_names` can be useful for readability of the exported model, although it would be possible to override them after the export.